pkg/util/arm/marshal.go

```diff
@@ -10,12 +10,37 @@ import (
 	"strings"
 
 	gofrsuuid "github.com/gofrs/uuid"
+
+	armnetwork "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v6"
 )
 
 const (
 	track2sdkPkgPathPrefix = "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager"
 )
 
+// ARM resource types that should include Type and Location fields when marshaling to JSON
+// Add new types here as needed.
+var armResourceTypes = map[reflect.Type]bool{
```

Comment:
I feel really strongly that we should not include such a list in the code. This is a nightmare to maintain. Let's instead look for a way to identify these types directly in `MarshalJSON()`.

What exactly is the difference between these types and others when marshalled?